src/main/java/org/broadinstitute/hellbender/utils/downsampling/Downsampler.java

```diff
@@ -20,6 +20,8 @@
      * immediately whether the item survives the downsampling process, while others will need to see
      * more items before making that determination.
      *
+     * Does not check its argument for null -- it is the client's responsibility not to submit null items.
```

Comment:
removing the check sets a bad precedent i think. Until we show that this check has a measurable performance impact, i think removing checks is not safe. What's stopping us from ditching these checks from everywhere? In profiling I've never seen those checks account for more than a very small fraction of runtime (< 1%)